docs-website/CONTRIBUTING.md

```diff
@@ -1,121 +1,178 @@
-# Contributing to Haystack Docs
+# Contributing to Haystack Documentation
```

Comment:
I would also link this page in https://github.com/deepset-ai/haystack/blob/main/CONTRIBUTING.md